### DISCUSSION TITLE
foundry-zksync installation fail over a wsl Ubuntu

### DISCUSSION THREAD

**Discussion by rpetitma** [Cyfrin/foundry-full-course-cu#2108] [Q&A] [answered]
Hi all!

I was trying to install the foundr-zksync Quick Install guide we all have in the repo but seems is not workins as expected:

At the end of the installation seems that everything goes fine and smooth

`foundryup-zksync: installing foundry (version nightly, tag nightly)
foundryup-zksync: downloading latest forge, and cast
################################################################################################################################################################################### 100.0%
foundryup-zksync: downloading manpages
################################################################################################################################################################################### 100.0%
foundryup-zksync: installed - forge 0.0.2 (825ca04 2024-07-22T10:16:47.022550197Z)
foundryup-zksync: installed - cast 0.0.2 (825ca04 2024-07-22T10:16:47.025711637Z)
foundryup-zksync: done!
Verifying installation...
Forge version 0.0.2 is successfully installed.`

But when trying to check if this is true... SURPRISE!

`@DESKTOP-VE0G57S:~/foundry-f23/foundry-zksync$` forge --version
forge 0.2.0 (0ad127a `2024-07-25T00:17:47.070402649Z)`

The new version should be 0.0.2 instead of 0.2.0

Did you guys faced something similar?

**Comment by EngrPips** [comment]
So I tried updating my `foundry-zksync` so as to check what version I will get if I run `forge --version`, and below is what I got.

<img width="1440" alt="Screenshot 2024-07-25 at 16 39 32" src="https://github.com/user-attachments/assets/871d7af0-9d3e-4571-98e3-bfd36c871e5a">

It seems to work fine for me.

**Comment by rpetitma** [comment]
Amazing... Run it again and now FAILS!
.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

 ╔═╗ ╔═╗ ╦ ╦ ╔╗╔ ╔╦╗ ╦═╗ ╦ ╦         Portable and modular toolkit
 ╠╣  ║ ║ ║ ║ ║║║  ║║ ╠╦╝ ╚╦╝    for Ethereum Application Development
 ╚   ╚═╝ ╚═╝ ╝╚╝ ═╩╝ ╩╚═  ╩                 written in Rust.

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

Fork of    : https://github.com/foundry-rs/
Repo       : https://github.com/foundry-zksync/
Book       : https://book.getfoundry.sh/                                     

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

foundryup-zksync: installing foundry (version nightly, tag nightly)
foundryup-zksync: downloading latest forge, and cast
################################################################################################################################################################################### 100.0%
foundryup-zksync: downloading manpages
################################################################################################################################################################################### 100.0%
foundryup-zksync: installed - forge 0.0.2 (825ca04 2024-07-22T10:16:47.022550197Z)
foundryup-zksync: installed - cast 0.0.2 (825ca04 2024-07-22T10:16:47.025711637Z)
foundryup-zksync: done!
**Verifying installation...
./install-foundry-zksync: line 38: forge: command not found
Installation verification failed. Forge is not properly installed.**

**Reply by EngrPips** [reply, marked as answer]
Can you uninstall and re-install?

**Reply by ArnaudBand** [reply]
I have the same issue, I tried to install `foundryup-zksync` but failed

```bash
foundryup-zksync
bash: foundryup-zksync: command not found
```

**Reply by EngrPips** [reply]
<img width="1440" alt="Screenshot 2024-07-26 at 08 27 28" src="https://github.com/user-attachments/assets/d5ec1a29-79d0-4c66-b806-ae1938a0b450">

It seems to work for me every time I run the command. @PatrickAlphaC  thought of informing you that people are having issues with installing `foundryzksync`

**Comment by rpetitma** [comment]
After few retries I've erased it all and start from scatch as @EngrPips pointed and WORKS!!! This is what I made to erase all and reinstall from scratch:

1. Remove the Foundry directory
Delete the ~/.foundry directory where the Foundry binaries are installed:

`**rm -rf ~/.foundry**`

2. Remove the foundry-zksync repository
If you have cloned the foundry-zksync repository, delete the directory where it is cloned:

**`rm -rf foundry-zksync`**

3. Remove PATH configurations
Edit your shell configuration file (~/.bashrc, ~/.zshrc, etc.) and remove any line that adds Foundry to the PATH. For example, look for and remove a line similar to this:

**`export PATH="$HOME/.foundry/bin:$PATH"`**
After editing the file, make sure to reload the shell configuration:

**`source ~/.bashrc  # or source ~/.zshrc`*

4. Verify that forge is no longer available
Ensure that the forge command is no longer available on your system:

**`which forge`**
This should return a message indicating that the command is not found.

5. Reinstall Foundry
Reinstall Foundry by following the official steps:

**`curl -L https://foundry.paradigm.xyz | bash`**
**`source ~/.bashrc  # or ~/.zshrc, depending on your shell`**
**`foundryup`**

Verify the installation:

**`forge --version`**

6. Reinstall foundry-zksync
Clone the foundry-zksync repository again and run the installation script:

**`git clone https://github.com/matter-labs/foundry-zksync.git`**
**`cd foundry-zksync`**
**`./install-foundry-zksync`**

Verify that the installed version of forge is correct:

**`forge --version`**

**Reply by EngrPips** [reply]
Awesome, You are awesome. I am glad you figured it out.

**Reply by rpetitma** [reply]
Many thanks!
